api/src/main/java/au/org/aodn/nrmn/restapi/service/SurveyIngestionService.java

```diff
@@ -40,32 +40,51 @@
     @Autowired
     ProgramRepository programRepository;
     @Autowired
+    SiteRepository siteRepo;
+    @Autowired
     EntityManager entityManager;
 
     public List<Observation> ingestStagedRow(StagedRowFormatted stagedRow) {
         return observationRepository.saveAll(getObservations(stagedRow));
     }
 
     public Survey getSurvey(StagedRowFormatted stagedRow) {
+        val site = stagedRow.getSite();
+
         val survey = Survey.builder()
                 .depth(stagedRow.getDepth())
                 .surveyNum(stagedRow.getSurveyNum().orElse(null))
-                .site(Site.builder().siteCode(stagedRow.getSite().getSiteCode()).build())
+                .site(Site.builder().siteCode(site.getSiteCode()).build())
                 .surveyDate(Date.valueOf(stagedRow.getDate()))
                 .build();
 
         Optional<Survey> existingSurvey = surveyRepository.findOne(Example.of(survey));
 
-        return existingSurvey.orElseGet(() -> surveyRepository.save(Survey.builder()
-                .depth(stagedRow.getDepth())
-                .surveyNum(stagedRow.getSurveyNum().orElse(null))
-                .direction(stagedRow.getDirection().toString())
-                .site(stagedRow.getSite())
-                .surveyDate(Date.valueOf(stagedRow.getDate()))
-                .surveyTime(Time.valueOf(stagedRow.getTime().orElse(null)))
-                .visibility(stagedRow.getVis().orElse(null))
-                .program(stagedRow.getRef().getStagedJob().getProgram())
-                .build()));
+        val siteSurveys = surveyRepository.findAll(
```

Comment:
Isn't enough just to set the sitecode to active if its not active - its being used on this survey - not sure why we are checking for existing survey's

Reply:
a site could be set to false trough  the data reference. not  only when it get created. 
So I had to check if was never used before

Reply:
Its not possible to set the site to be inactive or active  when adding or editing a site.   It gets defaulted to inactive when its created.

Reply:
I believe the intention of the isActive flag is to reflect whether there are any survey's associated with the site.  So the siteSurveys check below doesn't make any sense - we always want to set the site to be active if its inactive when creating a survey for that site.

@ccmoloney / @bpasquer can you clarify

Reply:
if it is then i will just check if it's false then

Reply:
I think the logic should be:

if isActive flag is FALSE
    If this is the first ingest for this site
        Set isActive flag to TRUE

Reply:
Doesn't isActive = false mean it is the first ingest for the site?

Reply:
So that flag is untrustworthy anyway.   Not sure why we're bothering.

Reply:
Only for new sites.  They may be able to deactivate old sites in the future